common/views/components/CopyUrl/CopyUrl.js

```diff
@@ -1,32 +1,103 @@
 // @flow
 
 import {spacing, font} from '../../../utils/classnames';
+import {Component} from 'react';
 import Icon from '../Icon/Icon';
 import HTMLInput from '../HTMLInput/HTMLInput';
 
 type Props = {|
   id: string,
-  url: string,
+  url: string
 |}
 
-const CopyUrl = ({id, url}: Props) => (
-  <div>
-    <HTMLInput
-      id='share'
-      type='text'
-      label='share url'
-      defaultValue={url}
-      isLabelHidden={true}
-      fontStyles={{s: 'HNL5', m: 'HNL4'}} />
-
-    <button aria-live='polite'
-      data-copy-text={url}
-      data-track-event={`{"category": "component", "action": "copy-url:click", "label": "id:${id}"}`}
-      className={`${spacing({s: 2}, {margin: ['top']})} ${font({s: 'HNM5', m: 'HNM4'})} btn btn--light is-hidden js-copy-url pointer`}>
-      <Icon name='check' extraClasses='icon--black is-hidden' />
-      <span className='js-copy-text'>Copy link</span>
-    </button>
-  </div>
-);
+type State = {|
+  isEnhanced: boolean,
```

Comment:
typo - and what is this?

Reply:
What's the typo?

I added `isEnhanced` to replicate our check for client side JS (so we e.g. wouldn't show the 'Copy URL' button if client JS isn't there).

I couldn't decide if this belonged at the component level, or if it should be higher up, where it should go.

Reply:
Feels like it's a hangover from the old site, so if we can keep it at the edge, and remove it later, that might be better.

Reply:
I'm on the verge of getting bogged down in whether or not we should be putting together a `withMustard` HOC or something similar, but realise that belongs in another PR. I'm going to leave this as-is, with a TODO to revisit when we decide if/how the mustard will be cut.